src/app/(proper_react)/redesign/(authenticated)/user/welcome/Onboarding.test.tsx

```diff
@@ -3,38 +3,133 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { it, expect } from "@jest/globals";
-import { render, screen } from "@testing-library/react";
+import { fireEvent, render, screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 import { composeStory } from "@storybook/react";
 import { axe } from "jest-axe";
 import Meta, { Onboarding } from "./Onboarding.stories";
 
+jest.mock("next/navigation", () => ({
+  useRouter: jest.fn(),
+}));
+
+function populateEnterInfoForm(container: HTMLElement) {
+  const firstNameInput = screen.getByPlaceholderText("Enter first name");
+  const lastNameInput = screen.getByPlaceholderText("Enter last name");
+  const locationInput = screen.getByPlaceholderText("Enter city and state");
+  const dateInput = container.querySelector("input[type='date']");
+
+  expect(firstNameInput).toBeInTheDocument();
+  expect(lastNameInput).toBeInTheDocument();
+  expect(locationInput).toBeInTheDocument();
+  expect(dateInput).toBeInTheDocument();
+
+  const firstName = "User";
+  const lastName = "Name";
+  const locationString = "New York, NY, USA";
+  const dateString = "2000-01-01";
+
+  fireEvent.change(firstNameInput, { target: { value: firstName } });
+  fireEvent.change(lastNameInput, { target: { value: lastName } });
+  fireEvent.change(locationInput, { target: { value: locationString } });
+
+  if (dateInput) {
```

Comment:
What does this line do?
Isn't `dateInput` always defined, and already `expect()`ed, per L25?

Reply:
It’s because the linter is not convinced that `dateInput` will be defined and I think that’s due to retrieving it with `.querySelector("input[type='date']")`.